Sources/AppTeamToolBox/MNQueueIterator.swift

```diff
@@ -43,12 +43,66 @@ public struct MNQueueIterator<Element: Identifiable> {
         fixIndexIntoRange()
     }
 
+    public mutating func insertAt(_ element: Element, index: Int) {
+        if index >= count {
+            items.append(element)
+        } else {
+            items.insert(element, at: index)
+        }
+        fixIndexIntoRange()
+    }
+
     public mutating func insertAtCurrentIndex(_ element: Element) {
         if currIndex == nil { currIndex = 0 }
         items.insert(element, at: currIndex!)
         fixIndexIntoRange()
     }
 
+    // Given a certain sorted array, this function determines whether to sort it based on a superset of a larger array, or an array simply in a different order.
+    public mutating func reorderItems(to elements: [Element], index: Int) {
+        // If the index is equal to 0, it indicates that the elements passed through as an argument is a larger array, and items is the subset of elements.
+        if index == -1 {
```

Comment:
Is this comment accurate? Seeing "if this index is equal to 0" immediately followed by `if index == -1` is kind of funky. What's going on here?

Reply:
That's a typo, I'll fix that comment